mobile/lib/models/file/file.dart

```diff
@@ -179,6 +181,15 @@ class EnteFile {
             hasExifTime = true;
             creationTime = exifTime.microsecondsSinceEpoch;
           }
+          mediaUploadData.isPanorama = checkPanoramaFromEXIF(null, exifData);
+
+          try {
+            final xmpData = XMPExtractor()
+                .extract(mediaUploadData.sourceFile!.readAsBytesSync());
+            mediaUploadData.isPanorama = checkPanoramaFromXMP(xmpData);
+          } catch (_) {}
+
+          mediaUploadData.isPanorama ??= false;
```

Comment:
These check should only happen for Normal Images (not for Live photo and video).
Also, if exifSays it's panorama, no need to check for xmp?

Reply:
For the first part, it is under a if where it only runs for file type image, hence file type live photo and video are eliminated.